firebase-app-distribution/src/test/java/com/google/firebase/appdistribution/FirebaseAppDistributionTesterApiClientTest.java

```diff
@@ -0,0 +1,111 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.appdistribution;
+
+import static androidx.test.InstrumentationRegistry.getContext;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertThrows;
+import static org.mockito.Mockito.when;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import javax.net.ssl.HttpsURLConnection;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+import org.robolectric.RobolectricTestRunner;
+
+@RunWith(RobolectricTestRunner.class)
+public class FirebaseAppDistributionTesterApiClientTest {
+
+  private static final String TEST_API_KEY = "AIzaSyabcdefghijklmnopqrstuvwxyz1234567";
+  private static final String TEST_APP_ID_1 = "1:123456789:android:abcdef";
+  private static final String TEST_AUTH_TOKEN = "fad.auth.token";
+  private static final String TEST_FID_1 = "cccccccccccccccccccccc";
+  private static final String INVALID_RESPONSE = "InvalidResponse";
+
+  private FirebaseAppDistributionTesterApiClient firebaseAppDistributionTesterApiClient;
+  @Mock private HttpsURLConnection mockHttpsURLConnection;
+
+  @Before
+  public void setup() throws Exception {
+    MockitoAnnotations.initMocks(this);
+
+    // using spy allows using doReturn to specify output
+    // of a method while leaving other methods unmocked
+    firebaseAppDistributionTesterApiClient =
+        Mockito.spy(new FirebaseAppDistributionTesterApiClient());
+
+    Mockito.doReturn(mockHttpsURLConnection)
+        .when(firebaseAppDistributionTesterApiClient)
+        .openHttpsUrlConnection(TEST_APP_ID_1, TEST_FID_1);
+  }
+
+  @Test
+  public void fetchLatestRelease_whenResponseSuccessful_returnsRelease() throws Exception {
+    JSONObject releaseJson = getTestJSON("testReleaseResponse.json");
+    InputStream response =
+        new ByteArrayInputStream(releaseJson.toString().getBytes(StandardCharsets.UTF_8));
+    when(mockHttpsURLConnection.getInputStream()).thenReturn(response);
+    AppDistributionRelease release =
+        firebaseAppDistributionTesterApiClient.fetchLatestRelease(
+            TEST_FID_1, TEST_APP_ID_1, TEST_API_KEY, TEST_AUTH_TOKEN);
+    assertEquals(release.getBinaryType(), BinaryType.APK);
+    assertEquals(release.getBuildVersion(), "3");
```

Comment:
Add a test case for when release build version is a string, like "1.0" or has spaces

Reply:
After looking into it and trying it out, the app distro build version returned from the endpoint is taken from the versionCode defined in the app's build.gradle which can only be an int. However, the versionName which becomes the display version in the release supports putting any string value, like "1.0 " which is being tested below, but not with spaces. Should there be one for spaces in the display version maybe?   

Sidenote, currently we are comparing the build versions to see if the tester is on the latest release but this will be changed to comparing the codeHash on a future PR.

Reply:
I think this code should be defensive around server errors (or changes), logically this should be an int, but should a bug happen, what do we want the SDK behavior to be?

Reply:
Spoke with Pranav and we added a catch for a NumberFormatException when parsing the Long fails, and he said that it would be better to add the test in another current PR we have up since some of the logic around that changed.